itkwidgets/cell_watcher.py

```diff
@@ -1,69 +1,96 @@
 import asyncio
 import sys
 from inspect import isawaitable, iscoroutinefunction
+from typing import Callable, Dict, List
 from IPython import get_ipython
+from IPython.core.interactiveshell import ExecutionResult
 from queue import Queue
 from imjoy_rpc.utils import FuturePromise
+from zmq.eventloop.zmqstream import ZMQStream
 
 background_tasks = set()
 
 
 class Viewers(object):
+    """This class is designed to track each instance of the Viewer class that
+    is instantiated as well as whether or not that instance is available for
+    updates or requests.
+    """
     def __init__(self):
         self._data = {}
 
     @property
-    def data(self):
-        return self._data
-
-    @property
-    def not_created(self):
-        # Return a list of names of viewers that have not been created yet
-        names = []
-        for key, val in self.data.items():
-            name = val['name']
-            if not val['status']:
-                name = name if name is not None else key
-                names.append(name)
-        return names
+    def data(self) -> Dict[str, Dict[str, bool]]:
+        """Get the underlying data dict containg all viewer data
 
-    @property
-    def not_named(self):
-        # Return a list of names of viewers that have not been named yet
-        return any([k for k, v in self.data.items() if v['name'] is None])
+        :return: The data object that contains all created Viewer information.
```

Comment:
can we describe here what the `str`, `str`, `bool`'s are?

Reply:
Sure thing!